### ISSUE TITLE
PayPal pay upon invoive unvailable due to Metadata header error

### ISSUE THREAD

**Issue by wesleyklose** [woocommerce/woocommerce-paypal-payments#1192]
### Describe the Bug

After filling out the required form for PayPal pay upon invoice an placing the order you get the presented this error message:

device data not_available Device data is not available for processing this order. The Paypal-Client-Metadata-Id header value sent during 'create Order' api call is either missing or incorrect or there was an error in collecting required data. please verify if appropiate value for PayPal-client-Metadata-Id header is being seint during 'Create Order' api call. Please note this error only applies to payment_source.pay_upon_invoice at the moment.

### To Reproduce

1. Select PayPal pay upon Invoice in as paymethod
2. fill out form and place order

### Screenshots
![Fehlercode](https://user-images.githubusercontent.com/48801679/219059501-9fa61b0d-84e2-4e00-8c26-b3395c2115f0.png)

### Expected Behavior
order gets placed

### Actual Behavior
see error code. order gets placed in shop but gets status failed after a few seconds.

### Environment

 - WordPress Version 6.1.1 uses the Theme Flatsome Child
 - WooCommerce Version 7.3.0
 - Plugin Version 2.0.2
 - Browser Chrome and Version 110.0.5481.77
 - Any other plugins installed - ~ 40 other plugins installed not all activated

### Additional Details
we currently have issued a ticket with Paypal TechSupport. Also PayOne said a rollback to a previous version should fix it but it didnt

<details><summary>System status</summary>

  ```
  <!--If applicable, paste the system status here. Please ensure you redact or remove any identifying information. -->
  ```

</details>

**Comment by InpsydeNiklas** [comment]
Hello @wesleyklose, the team has addressed this in issue #1177. 
You can try out a test version with this change from [this download link](https://drive.google.com/uc?export=download&id=1VjO-dYb2t6pcWdNF0dnb0pKlJEwLBPBE). 
We appreciate your feedback on how this version works for you.
Thanks!

**Comment by wesleyklose** [comment]
Hey Niklas,

installed the version and tested pay upon invoice. During Checkout I didn't
receive any errors.

However the placed order failed 6 minutes later because of a not received
webhook. "Could not process WC order because PAYMENT.CAPTURE.COMPLETED webhook not received."

**Comment by wesleyklose** [comment]
![Screenshot 2023-02-16 104942](https://user-images.githubusercontent.com/48801679/219332768-0af5e32d-23bc-4ada-9a7a-61a68d508fd6.png)
 here with actual picture

**Comment by wesleyklose** [comment]
@InpsydeNiklas any suggestions?

**Comment by wesleyklose** [comment]
@InpsydeNiklas did you have a chance to take a look at this already?

**Comment by InpsydeNiklas** [comment]
@wesleyklose The webhooks can be resubscribed from the [Webhook Status section in the Connection tab](https://woocommerce.com/document/woocommerce-paypal-payments/#webhook-status).
But please [get in touch with the support team from here](https://woocommerce.com/document/woocommerce-paypal-payments/#get-help) if you need additional guidance.
Thanks!

**Comment by InpsydeNiklas** [comment]
This issue was resolved in the last update [2.0.3.](https://github.com/woocommerce/woocommerce-paypal-payments/releases/tag/2.0.3)


> Fix - DEVICE_DATA_NOT_AVAILABLE error message when FraudNet is enabled https://github.com/woocommerce/woocommerce-paypal-payments/pull/1177

Any further questions will be clarified with the support team directly, so I will close this issue.

**Comment by patrick-7-7-7** [comment]
You seem to have reintroduced the issue with 2.0.4?
I installed this today (2.0.4), got the error, tested with 2.01 and 2.03, it worked fine there. Upgrading to 2.04 reintroduced the error. I did not test 2.02.

This shows up in the checkout and in the backend under "orders":

DEVICE_DATA_NOT_AVAILABLE Device Data is not available for processing this order. The PayPal-Client-Metadata-Id header value sent during `Create Order` api call is either missing or incorrect or there was an error in collecting required data. Please verify if appropriate value for PayPal-Client-Metadata-Id header is being sent during 'Create Order' api call. Please note this error only applies to payment_source.pay_upon_invoice at the moment.

Can you quickly reproduce it or do you need some proper info? I am not sure I can provide it.
